tests/pipeline/test_buyback_auth.py

```diff
@@ -0,0 +1,464 @@
+"""
+Tests for the reference loader for Buyback Authorizations.
+"""
+from functools import partial
+from unittest import TestCase
+
+import blaze as bz
+from blaze.compute.core import swap_resources_into_scope
+from contextlib2 import ExitStack
+from nose_parameterized import parameterized
+import numpy as np
+import pandas as pd
+from pandas.util.testing import assert_series_equal
+from six import iteritems
+
+from zipline.pipeline import Pipeline
+from zipline.pipeline.data import (CashBuybackAuthorizations,
+                                   ShareBuybackAuthorizations)
+from zipline.pipeline.engine import SimplePipelineEngine
+from zipline.pipeline.factors.events import (
+    BusinessDaysSincePreviousCashBuybackAuth,
+    BusinessDaysSincePreviousShareBuybackAuth
+)
+from zipline.pipeline.loaders.buyback_auth import \
+    CashBuybackAuthorizationsLoader, ShareBuybackAuthorizationsLoader
+from zipline.pipeline.loaders.blaze import (
+    BlazeCashBuybackAuthorizationsLoader,
+    BlazeShareBuybackAuthorizationsLoader,
+    BUYBACK_ANNOUNCEMENT_FIELD_NAME,
+    SHARE_COUNT_FIELD_NAME,
+    SID_FIELD_NAME,
+    TS_FIELD_NAME,
+    CASH_FIELD_NAME
+)
+from zipline.utils.numpy_utils import make_datetime64D, NaTD
+from zipline.utils.test_utils import (
+    gen_calendars,
+    make_simple_equity_info,
+    num_days_in_range,
+    tmp_asset_finder,
+)
+
+
+sids = A, B, C, D, E = range(5)
+
+equity_info = make_simple_equity_info(
+    sids,
+    start_date=pd.Timestamp('2013-01-01', tz='UTC'),
+    end_date=pd.Timestamp('2015-01-01', tz='UTC'),
+)
+
+buyback_authorizations = {
+    # K1--K2--A1--A2--SC1--SC2--V1--V2.
```

Comment:
Forgot to consistently update the comments for this section; but the SC field is there (share count) and the V is there, but it should be modified since it refers to the cash field.